### PR TITLE
Fix iterations, Part 2

### PR DESCRIPTION
**Pull request by michaelkleinhenz**
Fixes several more issues in the iterations panel:
- when the iteration on a work item changes, the counts on the iteration panel are updated.
- the root iteration is no longer displayed in the iteration list.
- several severe bug fixes in the mock data service(s).
- the date display is not visible when no dates are set (no lone "-" display on the current iteration in that case).

This also fixes some other things:
- the list is now updated correctly when selecting a new iteration (only in list view, in board view, this is still broken).
- fixes some other mock related problems.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -106,7 +106,7 @@
     "@angular/platform-browser-dynamic": "4.0.1",
     "@angular/platform-server": "4.0.1",
     "@angular/router": "4.0.1",
-    "angular-tree-component": "3.2.4",
+    "angular-tree-component": "3.2.3",
```
Comment (nimishamukherjee) [inline package.json]: Platform is using "angular-tree-component": "3.2.4"

Comment (michaelkleinhenz) [inline package.json]: Ah, yes. Have to check that again. 3.2.4 was broken.

**Comment by michaelkleinhenz** [comment]
Weird, I checked the integrated variant, and it was working. Will check.

**Comment by michaelkleinhenz** [comment]
@nimishamukherjee ok, the core reports wrong numbers currently. @pranavgore09 already has a PR in place to fix them. This PR is displaying the "correct" numbers reported by the core. So all ok here.

**Comment by michaelkleinhenz** [comment]
@nimishamukherjee on the tree component: the problem there is still not fixed. We either break the tests or prod with the new tree component version.

**Comment by fabric8cd** [comment]
@michaelkleinhenz snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1561.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@michaelkleinhenz snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1561.int.rdu2c.fabric8.io

**Pull request merged**